workflow/workflow_test.go

```diff
@@ -798,3 +805,114 @@ func TestEarlyContextCancellation(t *testing.T) {
 		t.Fatalf("Test execution time is greater than 1000 milliseconds; Is the workflow properly cancelling?")
 	}
 }
+
+var justMathWorkflowDefinition = `
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        a:
+         type:
+           type_id: integer
+        b:
+         type:
+           type_id: integer
+steps:
+  no_wait:
+    plugin:
+      src: "n/a"
+      deployment_type: "builtin"
+    step: wait
+    input:
+      wait_time_ms: 0
+outputs:
+  success:
+    result: !expr intToString($.input.a + $.input.b) + " and " + $.steps.no_wait.outputs.success.message
+`
+
+func TestWorkflowWithMathAndFunctions(t *testing.T) {
+	// For this test, we run the minimum amount of steps, and resolve the output with math and functions.
+	logConfig := log.Config{
+		Level:       log.LevelDebug,
+		Destination: log.DestinationStdout,
+	}
+	logger := log.New(
+		logConfig,
+	)
+	cfg := &config.Config{
+		Log: logConfig,
+	}
+	stepRegistry := NewTestImplStepRegistry(logger, t)
+
+	executor := lang.Must2(workflow.NewExecutor(
+		logger,
+		cfg,
+		stepRegistry,
+		builtinfunctions.GetFunctions(),
+	))
+	wf := lang.Must2(workflow.NewYAMLConverter(stepRegistry).FromYAML([]byte(justMathWorkflowDefinition)))
+	preparedWorkflow := lang.Must2(executor.Prepare(wf, map[string][]byte{}))
+	outputID, outputData, err := preparedWorkflow.Execute(context.Background(), map[string]any{
+		"a": int64(2),
+		"b": int64(2),
+	})
+	assert.NoError(t, err)
+	assert.Equals(t, outputID, "success")
+	assert.Equals(t, outputData.(map[any]any), map[any]any{
+		"result": "4 and Plugin slept for 0 ms.",
```

Comment:
The message sure looks odd, but I guess it proves your point. 😆

Reply:
Yeah. It wasn't worth adding to the test dependencies to have a prettier output.